tests/Builder/Accumulator/AvgAccumulatorTest.php

```diff
@@ -0,0 +1,61 @@
+<?php
+
+declare(strict_types=1);
+
+namespace MongoDB\Tests\Builder\Accumulator;
+
+use MongoDB\Builder\Accumulator;
+use MongoDB\Builder\Expression;
+use MongoDB\Builder\Pipeline;
+use MongoDB\Builder\Stage;
+use MongoDB\Tests\Builder\PipelineTestCase;
+
+use function MongoDB\object;
+
+/**
+ * Test $avg accumulator
+ */
+class AvgAccumulatorTest extends PipelineTestCase
+{
+    public function testUseInGroupStage(): void
+    {
+        $pipeline = new Pipeline(
+            Stage::group(
+                _id: Expression::fieldPath('item'),
+                avgAmount: Accumulator::avg(
+                    Expression::multiply(
+                        Expression::numberFieldPath('price'),
+                        Expression::intFieldPath('quantity'),
+                    ),
+                ),
+                avgQuantity: Accumulator::avg(
+                    Expression::intFieldPath('quantity'),
+                ),
+            ),
+        );
+
+        $this->assertSamePipeline(Pipelines::AvgUseInGroupStage, $pipeline);
+    }
+
+    public function testUseInSetWindowFieldsStage(): void
+    {
+        $pipeline = new Pipeline(
+            Stage::setWindowFields(
+                partitionBy: Expression::fieldPath('state'),
+                sortBy: object(
+                    orderDate: 1,
+                ),
+                output: object(
```

Comment:
I'm always using the same way to write the pipelines (using `object` and not `array` map), so I don't really cover all the feature.